File: src/Federation.App/ClientApp/src/models/players/uscf/uscfPlayerTournament.ts
export class UscfPlayerTournament {
  eventId: string;
  eventName: string;
  eventState: string;
  sectionName: string;
  endDate: Date;

  private constructor(
    eventId: string,
    eventName: string,
    eventState: string,
    sectionName: string,
    endDate: Date,
  ) {
    this.eventId = eventId;
    this.eventName = eventName;
    this.eventState = eventState;
    this.sectionName = sectionName;
    this.endDate = new Date(endDate);
  }

  static fromObject<T extends UscfPlayerTournament = UscfPlayerTournament>(
    obj: T,
  ) {
    return new UscfPlayerTournament(
      obj.eventId,
      obj.eventName,
      obj.eventState,
      obj.sectionName,
      obj.endDate,
    );
  }
}

File: src/Federation.Services.Uscf/Models/UscfPlayerTournament.cs
namespace Federation.Services.Uscf.Models;

public class UscfPlayerTournament
{
#nullable disable
    public string EventId { get; set; }

    public string EventName { get; set; }

    public string SectionName { get; set; }
#nullable enable

    public DateTime EndDate { get; set; }
}